test_prebuild/test_track_variables.py

```diff
@@ -0,0 +1,142 @@
+#! /usr/bin/env python3
+"""
+-----------------------------------------------------------------------
+ Description:  Contains unit tests for ccpp_track_variables.py script
+
+ Assumptions:  Assumes user has correct environment for running ccpp_track_variables.py script.
+               This script should not be run directly, but rather invoked with pytest.
+
+ Command line arguments: none
+
+ Usage: pytest test_track_variables.py         # run the unit tests
+-----------------------------------------------------------------------
+"""
+import sys
+import os
+import pytest
+
+TEST_DIR = os.path.dirname(os.path.abspath(__file__))
+SCRIPTS_DIR = os.path.abspath(os.path.join(TEST_DIR, os.pardir, "scripts"))
+SAMPLE_FILES_DIR = "test_track_variables"
+SUITE_FILE = f'{SAMPLE_FILES_DIR}/suite_TEST_SUITE.xml'
+SMALL_SUITE_FILE = f'{SAMPLE_FILES_DIR}/suite_small_suite.xml'
+CONFIG_FILE = f'{SAMPLE_FILES_DIR}/ccpp_prebuild_config.py'
+if not os.path.exists(SCRIPTS_DIR):
+    raise ImportError(f"Cannot find scripts directory {SCRIPTS_DIR}")
+
+sys.path.append(SCRIPTS_DIR)
+
+from ccpp_track_variables import track_variables
+
+def test_successful_match(capsys):
+    """Tests whether test_track_variables.py produces expected output from sample suite and
+       metadata files for a case with a successful match (user provided a variable that exists
+       within the schemes specified by the test suite)"""
+    expected_output = """For suite test_track_variables/suite_small_suite.xml, the following schemes (in order for each group) use the variable air_pressure:
+In group group1
+  scheme_1_run (intent in)
+  scheme_1_run (intent in)"""
+    track_variables(SMALL_SUITE_FILE,SAMPLE_FILES_DIR,CONFIG_FILE,'air_pressure',False)
+    streams = capsys.readouterr()
+    expected_output_list = expected_output.splitlines()
+    streams_err_list = streams.err.splitlines()
+    # Split into lines to ignore whitespace differences
+    i = 0
+    for line in streams_err_list:
+        assert streams_err_list[i].strip() == expected_output_list[i].strip()
+        i+=1
```

Comment:
Would it make sense to use:
```
for (err, expected) in zip(streams_err_list, expected_output_list):
    assert err.strip() == expected.strip()
```

Reply:
Thanks for the suggestion, this is a much cleaner implementation for sure.